src/natcap/invest/urban_cooling_model.py

```diff
@@ -856,12 +854,12 @@ def calculate_uhi_result_vector(
     target_uhi_layer.StartTransaction()
     for feature in target_uhi_layer:
         feature_id = feature.GetFID()
-        if feature_id in cc_stats and cc_stats[feature_id]['count'] > 0:

```

Comment:
Oh that's interesting. That reminded me of https://github.com/natcap/invest/issues/64. I think we usually assume that all features will have a geometry, and don't handle the case where a geometry is missing. So maybe it's okay to put off until we address #64? If the current behavior is at least consistent with other models.

Reply:
Sorry @emlys , by "okay to put off" do you mean going with the original code or going with your changes?

> I think we usually assume that all features will have a geometry, and don't handle the case where a geometry is missing. 

I think that's right. Although when a missing geometry does crash a model and someone reports it on the forum, we tend to update our code to tolerate the missing geometry without crashing. So in that sense, I think the original code here is a bit safer and where we might end up anyway. 

Unless there's another reason -- other than missing geometry -- that would make us want that to see the `KeyError`.

Reply:
> Unless there's another reason -- other than missing geometry -- that would make us want that to see the KeyError.

I was suggesting going with these changes, the reason being that this is one of the models that assumes FIDs are the same across an intermediate and output vector. That seems to work fine, but I'm not confident saying it will _always_ work. Seeing that KeyError would let us know if there's a problem. 

Also, most of the models that use `pygeoprocessing.zonal_statistics` like this do not allow missing FIDs - all except urban cooling and urban flood would raise a KeyError. If we're leaning towards allowing null geometries (#64), maybe all the others should be updated to allow that?

Reply:
>I was suggesting going with these changes, the reason being that this is one of the models that assumes FIDs are the same across an intermediate and output vector. That seems to work fine, but I'm not confident saying it will always work. Seeing that KeyError would let us know if there's a problem.

Makes sense, let's go with it.